plugins/database/mongodb/mongodb_test.go

```diff
@@ -233,3 +237,191 @@ func testCreateDBUser(t testing.TB, connURL, db, username, password string) {
 		t.Fatal(result.Err())
 	}
 }
+
+func TestRedact(t *testing.T) {
+	type testCase struct {
+		input    dbplugin.DatabaseConfig
+		expected dbplugin.DatabaseConfig
+	}
+
+	tests := map[string]testCase{
+		"empty returns empty": {
+			input:    dbplugin.DatabaseConfig{},
+			expected: dbplugin.DatabaseConfig{},
+		},
+		"non-redacted are unaffected": {
+			input: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"foo":        "bar",
+					"some thing": "omgwtfbbq",
+				},
+			},
+			expected: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"foo":        "bar",
+					"some thing": "omgwtfbbq",
+				},
+			},
+		},
+		"tls_certificate_key": {
+			input: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"tls_certificate_key": "foobar",
+				},
+			},
+			expected: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"tls_certificate_key": "*****",
+				},
+			},
+		},
+		"tls_ca": {
+			input: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"tls_ca": "foobar",
+				},
+			},
+			expected: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"tls_ca": "*****",
+				},
+			},
+		},
+		"tls_certificate_key and tls_ca": {
+			input: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"tls_certificate_key": "foobar",
+					"tls_ca":              "foobar",
+				},
+			},
+			expected: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"tls_certificate_key": "*****",
+					"tls_ca":              "*****",
+				},
+			},
+		},
+		"mixed values": {
+			input: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"foo":                 "bar",
+					"some thing":          "omgwtfbbq",
+					"tls_certificate_key": "foobar",
+					"tls_ca":              "foobar",
+				},
+			},
+			expected: dbplugin.DatabaseConfig{
+				ConnectionDetails: map[string]interface{}{
+					"foo":                 "bar",
+					"some thing":          "omgwtfbbq",
+					"tls_certificate_key": "*****",
+					"tls_ca":              "*****",
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			m := new()
+			actual := m.Redact(test.input)
+			if !reflect.DeepEqual(actual, test.expected) {
+				t.Fatalf("Redaction failed:\nActual:\n%#v\nExpected:%#v", actual, test.expected)
+			}
+		})
+	}
+}
+
+func TestGetTLSAuth(t *testing.T) {
+	// certs := MakeCerts(t)
```

Comment:
```suggestion
```

Reply:
Refactored as a part of the integration test